perses/distributed/relative_setup.py

```diff
@@ -399,8 +399,14 @@ def __init__(self, topology_proposal, pos_old, new_positions, use_dispersion_cor
         a_0, b_0, c_0, alpha_0, beta_0, gamma_0 = mdtrajutils.unitcell.box_vectors_to_lengths_and_angles(*self._lambda_zero_sampler_state.box_vectors)
         a_1, b_1, c_1, alpha_1, beta_1, gamma_1 = mdtrajutils.unitcell.box_vectors_to_lengths_and_angles(*self._lambda_one_sampler_state.box_vectors)
 
-        self._lambda_zero_traj = md.Trajectory(np.array(self._lambda_zero_sampler_state.positions), self._factory.hybrid_topology, unitcell_lengths=[a_0, b_0, c_0], unitcell_angles=[alpha_0, beta_0, gamma_0])
-        self._lambda_one_traj = md.Trajectory(np.array(self._lambda_one_sampler_state.positions), self._factory.hybrid_topology, unitcell_lengths=[a_1, b_1, c_1], unitcell_angles=[alpha_1, beta_1, gamma_1])
+        #subset the topology appropriately:
+        atom_selection = self._factory.hybrid_topology.select("not water")
+        subset_topology = self._factory.hybrid_topology.subset(atom_selection)
+        lambda_zero_positions = self._lambda_zero_sampler_state.positions[atom_selection, :]
+        lambda_one_positions = self._lambda_one_sampler_state.positions[atom_selection, :]
+
+        self._lambda_zero_traj = md.Trajectory(np.array(lambda_zero_positions), subset_topology, unitcell_lengths=[a_0, b_0, c_0], unitcell_angles=[alpha_0, beta_0, gamma_0])
```

Comment:
Don't implicitly strip units from `sampler_state.positions` this way---explicitly convert them to MDTraj units.